gridcapa-cse-valid-app/src/main/java/com/farao_community/farao/cse_valid/app/TcDocumentTypeWriter.java

```diff
@@ -184,64 +191,111 @@ public void fillNoTtcAdjustmentError(CseValidRequest cseValidRequest) {
+        // Temporary method : should be replaced with real handling of export-corner case
+        fillTimestampSuccess(initialTs, (ts, value) -> ts.setMIEC(buildQuantityType(value)), miecValue);
+    }
+
+    private void fillTimestampSuccess(TTimestamp initialTs, BiConsumer<TTimestamp, BigDecimal> valueTimestampFiller, BigDecimal value) {
         fillEmptyValidationResults();
         List<TTimestamp> listTimestamps = tcDocumentType.getValidationResults().get(0).getTimestamp();
-        TTimestamp ts = new TTimestamp();
-        ts.setReferenceCalculationTime(initialTs.getReferenceCalculationTime());
+        TTimestamp ts = initializeNewTimestampWithExistingTimeData(initialTs);
 
-        QuantityType mnii = new QuantityType();
-        mnii.setV(initialTs.getMiBNII().getV().subtract(initialTs.getANTCFinal().getV()));
+        valueTimestampFiller.accept(ts, value);
 
-        completeFillingWithStatusSuccess(ts, initialTs, mnii);
+        completeFillingWithStatusSuccess(ts, initialTs);
+        ts.setBASECASEfile(initialTs.getBASECASEfile());
 
         listTimestamps.add(ts);
     }
 
-    public void fillTimestampNoVerificationNeeded(TTimestamp initialTs) {
+    public void fillTimestampWithDichotomyResponse(TTimestamp initialTs, BigDecimal mibniiValue, BigDecimal mniiValue, TLimitingElement tLimitingElement) {
         fillEmptyValidationResults();
         List<TTimestamp> listTimestamps = tcDocumentType.getValidationResults().get(0).getTimestamp();
-        TTimestamp ts = new TTimestamp();
-        ts.setReferenceCalculationTime(initialTs.getReferenceCalculationTime());
+        TTimestamp ts = initializeNewTimestampWithExistingTimeData(initialTs);
 
-        QuantityType mnii = new QuantityType();
-        mnii.setV(initialTs.getMNII().getV());
+        ts.setMiBNII(buildQuantityType(mibniiValue));
+        ts.setMNII(buildQuantityType(mniiValue));
+
+        completeFillingWithStatusSuccess(ts, initialTs);
+        ts.setLimitingElement(tLimitingElement); // override initial value set in completeFillingWithStatusSuccess by default
+
+        listTimestamps.add(ts);
+
+        listTimestamps.sort(Comparator.comparing(c -> OffsetDateTime.parse(c.getTime().getV())));
+    }
+
+    public void fillDichotomyError(TTimestamp initialTs) {
+        fillEmptyValidationResults();
+        List<TTimestamp> listTimestamps = tcDocumentType.getValidationResults().get(0).getTimestamp();
+        TTimestamp ts = initializeNewTimestampWithExistingTimeData(initialTs);
 
-        completeFillingWithStatusSuccess(ts, initialTs, mnii);
+        completeFillingWithError(BigInteger.ONE, ts, ERROR_MSG_GENERIC);
 
         listTimestamps.add(ts);
     }
 
-    private void completeFillingWithStatusSuccess(TTimestamp ts, TTimestamp initialTs, QuantityType mnii) {
+    private static QuantityType buildQuantityType(BigDecimal mibniiValue) {
+        QuantityType mibnii = new QuantityType();
+        mibnii.setV(mibniiValue);
+        return mibnii;
+    }
```

Comment:
You can rename the variables to make it generic

Reply:
Exact